docs/pages/guides/best-practices/dividing-into-systems.mdx

```diff
@@ -0,0 +1,161 @@
+The root namespace is [extremely privileged](/guides/best-practices/system-best-practices#avoid-the-root-namespace-if-possible).
+So the code that has to run the root namespace should be a separate `System` from the code that can be in a different namespace.
+
+## Bad reasons
+
+There are problems that could be solved by dividing your logic into multiple `System`s, but that are better handled by other solutions.
+
+### Shared logic
+
+If some logic is shared between two different `System`s, it is tempting to write a third `System` that implements it and call it from both of them.
+However, a simpler solution is to use a [Solidity library](https://solidity-by-example.org/library/) to implement the shared logic.
+
+### Contract size limit
+
+Ethereum contracts are limited in size, but you can always use [a public library](/guides/best-practices/system-best-practices#use-libraries-to-bypass-the-contract-size-limit) to work around that limit.
+
+## Calling multiple `System`s
+
+It is pretty common to already have two different `System`s, and then need to call them together.
+For example, imagine you have a `System` that manages the player's position, called `LocationSystem`.
+The player can call a `move` function.
+Then you have another `System`, `EnergySystem`.
+The player can call an `eat` function.
+Now you want a combined action, `eatAndMove`.
+
+There are multiple ways to create this combined action, only some of which require the extra complication of a third `System`.
+
+### Use batch calls (from the client)
+
+Under these conditions:
+
+- You want `eat` and `move` to happen in sequence.
+- You want them to happen atomically (so you can't just call `eat` and then `move`).
+- You have no additional logic that needs to happen between them.
+
+You can achieve this by using [batch calls](/world/batch-calls), without any need for changes in the `World`.
+
+<details>
+<summary>Illustration</summary>
+![](./batchCall.svg)
+
+1. The client sends a [`batchCall`](/world/batch-calls) that includes call to `eat` and `move`.
+1. The `batchCall` component inside the `World` calls the first function, `eat`.
+1. After the `eat` function returns, the `batchCall` component calls `move`.
+
+</details>
+
+### Move code into libraries
+
+Under these conditions:
+
+- You control the namespace of `LocationSystem` and `EnergySystem`.
+- The permissions required for `move` and `eat` are the same.
+
+You can move the code for `move` and `eat` into [public libraries](/guides/best-practices/system-best-practices#use-libraries-to-bypass-the-contract-size-limit).
+These libraries can then be called from `LocationSystem`, `EnergySystem`, and `CombinedActionSystem`.
+
+<details>
+  <summary>Illustration</summary>
+  ![](./libraries.svg)
+</details>
+
+Alternatively, you might be able to unite `LocationSystem` and `EnergySystem` into a single `GameSystem`, which turns the calls from `eatAndMove` into standard function calls within a contract.
```

Comment:
Mhm I don't feel good about this pattern (see comment above) but I understand what you're trying to point out. Maybe we can rephrase to be more like "if the logic is always coupled (ie moving always takes energy or sth) then you should combine them", but then it's basically the same argument as in "Modularity", so I maybe we should just remove this sentence.